src/components/Template/Template.tsx

```diff
@@ -1,23 +1,65 @@
 /** @jsx h */
 
 import { h, Component } from 'preact';
-import PropTypes from 'prop-types';
 import { renderTemplate, isEqual } from '../../lib/utils';
 
-class Template extends Component {
-  shouldComponentUpdate(nextProps) {
+type TemplateProps = {
+  data?: Record<string, any>;
+  rootProps?: Record<string, any>;
+  rootTagName?: string;
+  templateKey?: string;
+  templates?: Record<
+    string,
+    | string
+    | ((data: any, bindEvent: (...args: any[]) => string) => string)
+    | undefined
+  >;
```

Comment:
```suggestion
  templates?: Record<
    string,
    Template<TData>
  >;
```

and passing Data to the props too?

Reply:
(Template should come from /types/widget)